Changed APIEndpoint.Port from int to int32

Xenwar opened the pull request:
Changed APIEndpoint.Port from int to int32

Fixes #1418

tahsinrahman commented:
/ok-to-test

detiber commented:
/hold
holding since this is a breaking API change, we should not merge this until we are ready to create a release branch for v1alpha2 related features and introduce breaking changes to master.

ncdc commented:
v1alpha3 is starting to take shape!
/milestone v0.3.0

vincepri commented:
@Xenwar Would you be able to rebase on master and make the change against the `v1alpha3` types?

Xenwar commented:
> @Xenwar Would you be able to rebase on master and make the change against the `v1alpha3` types?

Done. Would please check.

vincepri commented on api/v1alpha2/cluster_types.go:
```diff
@@ -139,7 +139,7 @@ type APIEndpoint struct {
 
 	// The port on which the API server is serving.
-	Port int `json:"port"`
+	Port int32 `json:"port"`
```
This should stay `int`

Xenwar replied:
Done

vincepri commented:
@Xenwar Seems you might need to run `make generate`

Xenwar commented:
> @Xenwar Seems you might need to run `make generate`

No familiar with generate. 
of the following, which file do I need to add
```
	modified:   config/crd/bases/cluster.x-k8s.io_clusters.yaml
	modified:   go.mod

```
